dali/test/python/test_pipeline.py

```diff
@@ -1774,3 +1774,25 @@ def test_pipeline_wrong_device_id():
         pipe.build()
         pipe.run()
     assert "device_id" in str(x.exception).lower()
+
+def test_properties():
+    @dali.pipeline_def(batch_size=11, prefetch_queue_depth={"cpu_size":3, "gpu_size":2})
+    def my_pipe():
+        pipe = Pipeline.current()
```

Comment:
Do we advertise `Pipeline.current()` anywhere?

Reply:
There's a whole paragraph on it in `pipeline.rst`
```
Current Pipeline
----------------

Subgraphs that do not contribute to the pipeline output are automatically pruned.
If an operator has side effects (e.g. ``PythonFunction`` operator family), it cannot be invoked
without setting the current pipeline. Current pipeline is set implicitly when the graph is
defined inside derived pipelines' :meth:`Pipeline.define_graph` method.
Otherwise, it can be set using context manager (``with`` statement)::

    pipe = dali.Pipeline(batch_size=N, num_threads=3, device_id=0)
    with pipe:
        src = dali.ops.ExternalSource(my_source, num_outputs=2)
        a, b = src()
        pipe.set_outputs(a, b)

When creating a pipeline with :meth:`pipeline_def`, the function which defines the pipeline is
executed within the scope of the newly created pipeline. The following example is equivalent
to the previous one::

    @dali.pipeline_def(batch_size=N, num_threads=3, device_id=0)
    def my_pipe(my_source):
        return dali.fn.external_source(my_source, num_outputs=2)

    pipe = my_pipe(my_source)
```

Reply:
Maybe in getting started?
@awolant ?